File: backoffice/src/table/domain/TableStatus.py
"""
 *
 * Libraries 
 *
"""

from src.shared.domain            import IntValueObject
from .InvalidTableStatusException import InvalidTableStatusException

"""
 *
 * Classes 
 *
"""

class TableStatus( IntValueObject ):

    """
     *
     * Consts 
     *
    """

    __ENABLED = 1
    __DELETED = 2

    """
     *
     * Methods 
     *
    """
    
    def __init__( self, value : int ) -> None:
        super().__init__( value )
        if not self.isValid():
            raise InvalidTableStatusException( value )
    
    @classmethod
    def enabled( cls ): # -> ProductStatus
        self = cls( cls.__ENABLED )
        return self
    
    @classmethod
    def deleted( cls ): # -> ProductStatus
        self = cls( cls.__DELETED )
        return self
    
    def isValid( self ) -> bool :
        if self.equals( self.__DELETED ) or self.equals( self.__ENABLED ):
            return True
        return False

File: attention/src/shared/table/domain/valueObject/TableStatus.cs
/* 
 * 
 * Libraries
 *
*/

using attention.src.shared.domain;

/* 
 * 
 * Classes 
 *
*/

namespace attention.src.table.domain {

    public class TableStatus : IntValueObject {

        /* 
         * 
         * Constants
         *
        */

        private const int ENABLED = 1;
        private const int DELETED = 2;
        private const int BUSY    = 3;

        /* 
         * 
         * Methods
         *
        */

        public TableStatus( int value ) : base( value ) {
            if( !isValid() ) {
                throw new InvalidTableStatus( value );
            }
        }

        public bool isValid() {
            return equals( ENABLED ) || equals( DELETED ) || equals( BUSY );
        }

        public static TableStatus enabled() {
            return new TableStatus( ENABLED );
        }

        public static TableStatus deleted() {
            return new TableStatus( DELETED );
        }

        public static TableStatus busy() {
            return new TableStatus( BUSY );
        }

    }

}